### ISSUE TITLE
enforce sql_mode=only_full_group_by in the mysql that is built into ranche-server

### ISSUE THREAD

**Issue by dbuschman7** [rancher/rancher#3631]
I have a brand new Rancher(v0.59.0) install on AWS with SSL term setup. I have 1 RancherOS(rancheros-v0.4.2-hvm-0 (ami-53045239)) based host registered via the command line on the host using the "custom" command generator. The host comes up fine and displays in the UI.  Then, when I try to add a service, the network agent begins to cycle up for 11 seconds and then fail, repeating this beahvior forever. 

This looks to be similiar behavoir to Issue - https://github.com/rancher/rancher/issues/2024

The error in the from the network agent container("47a9cb32-6111-4105-b59f-55c679ec09fd" in my case ) is : 

[config.sh:148] error 'Missing SHA1SUMS files, invalid download'
[scripts.sh:24] echo ERROR: 'Missing SHA1SUMS files, invalid download'
ERROR: Missing SHA1SUMS files, invalid download

The corresponding errors on the rancher server are : 

2016-02-16 20:11:23,998 ERROR [:] [] [] [] [tp1085560730-18] [i.g.i.g.r.handler.ExceptionHandler  ] Exception in API for request [io.github.ibuildthecloud.gdapi.request.ApiRequest@65417298] org.jooq.exception.DataAccessException: SQL [select `host_0`.`id` as `host_0_id`, `host_0`.`name` as `host_0_name`, `host_0`.`account_id` as `host_0_account_id`, `host_0`.`kind` as `host_0_kind`, `host_0`.`uuid` as `host_0_uuid`, `host_0`.`description` as `host_0_description`, `host_0`.`state` as `host_0_state`, `host_0`.`created` as `host_0_created`, `host_0`.`removed` as `host_0_removed`, `host_0`.`remove_time` as `host_0_remove_time`, `host_0`.`data` as `host_0_data`, `host_0`.`uri` as `host_0_uri`, `host_0`.`compute_free` as `host_0_compute_free`, `host_0`.`compute_total` as `host_0_compute_total`, `host_0`.`agent_id` as `host_0_agent_id`, `host_0`.`zone_id` as `host_0_zone_id`, `host_0`.`physical_host_id` as `host_0_physical_host_id`, `host_0`.`is_public` as `host_0_is_public`, `host_0`.`agent_state` as `host_0_agent_state`, `ip_address_1`.`id` as `ip_address_1_id`, `ip_address_1`.`name` as `ip_address_1_name`, `ip_address_1`.`account_id` as `ip_address_1_account_id`, `ip_address_1`.`kind` as `ip_address_1_kind`, `ip_address_1`.`uuid` as `ip_address_1_uuid`, `ip_address_1`.`description` as `ip_address_1_description`, `ip_address_1`.`state` as `ip_address_1_state`, `ip_address_1`.`created` as `ip_address_1_created`, `ip_address_1`.`removed` as `ip_address_1_removed`, `ip_address_1`.`remove_time` as `ip_address_1_remove_time`, `ip_address_1`.`data` as `ip_address_1_data`, `ip_address_1`.`address` as `ip_address_1_address`, `ip_address_1`.`subnet_id` as `ip_address_1_subnet_id`, `ip_address_1`.`network_id` as `ip_address_1_network_id`, `ip_address_1`.`is_public` as `ip_address_1_is_public`, `ip_address_1`.`role` as `ip_address_1_role`, `ip_address_1`.`hostname` as `ip_address_1_hostname`, `ip_address_1`.`ip_pool_id` as `ip_address_1_ip_pool_id` from `ip_address` as `ip_address_1` join `host_ip_address_map` on `host_ip_address_map`.`ip_address_id` = `ip_address_1`.`id` join `host` as `host_0` on `host_0`.`id` = `host_ip_address_map`.`host_id` left outer join `instance_host_map` on `host_0`.`id` = `instance_host_map`.`host_id` where (`host_0`.`removed` is null and `host_0`.`state` not in (?, ?) and 1 = 1 and `ip_address_1`.`removed` is null and `host_0`.`account_id` = ?) group by `host_0`.`id`]; Expression #20 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'cattle.ip_address_1.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
	at org.jooq.impl.Utils.translate(Utils.java:1287) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:495) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:326) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:330) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.SelectImpl.fetch(SelectImpl.java:2256) ~[jooq-3.3.0.jar:na]
	at io.cattle.platform.configitem.context.dao.impl.MetaDataInfoDaoImpl.getInstanceHostMetaData(MetaDataInfoDaoImpl.java:135) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.context.impl.ServiceMetadataInfoFactory.getFullMetaData(ServiceMetadataInfoFactory.java:121) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.context.impl.ServiceMetadataInfoFactory.populateContext(ServiceMetadataInfoFactory.java:75) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.context.impl.AbstractAgentBaseContextFactory.populateContext(AbstractAgentBaseContextFactory.java:69) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.server.model.impl.AbstractArchiveBasedConfigItem.handleRequest(AbstractArchiveBasedConfigItem.java:56) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.server.impl.ConfigItemServerImpl.handleDownload(ConfigItemServerImpl.java:97) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.server.impl.ConfigItemServerImpl.handleRequest(ConfigItemServerImpl.java:40) ~[cattle-config-item-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.api.manager.ConfigContentManager.handle(ConfigContentManager.java:80) ~[cattle-config-item-api-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.configitem.api.manager.ConfigContentManager.getByIdInternal(ConfigContentManager.java:49) ~[cattle-config-item-api-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractBaseResourceManager.getById(AbstractBaseResourceManager.java:61) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.ResourceManagerRequestHandler.generate(ResourceManagerRequestHandler.java:49) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.AbstractResponseGenerator.handle(AbstractResponseGenerator.java:14) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.handle(DefaultReadWriteApiDelegate.java:27) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.DefaultReadWriteApiDelegate.read(DefaultReadWriteApiDelegate.java:17) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at sun.reflect.GeneratedMethodAccessor359.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:96) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:260) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:94) ~[spring-tx-3.2.4.RELEASE.jar:3.2.4.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204) ~[spring-aop-3.2.5.RELEASE.jar:3.2.5.RELEASE]
	at com.sun.proxy.$Proxy45.read(Unknown Source) ~[na:na]
	at io.github.ibuildthecloud.gdapi.request.handler.write.ReadWriteApiHandler.handle(ReadWriteApiHandler.java:21) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.github.ibuildthecloud.gdapi.servlet.ApiRequestFilterDelegate.doFilter(ApiRequestFilterDelegate.java:86) ~[cattle-framework-java-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter$1.runInContext(ApiRequestFilter.java:90) [cattle-framework-api-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.api.servlet.ApiRequestFilter.doFilter(ApiRequestFilter.java:83) [cattle-framework-api-0.5.0-SNAPSHOT.jar:na]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:364) [jetty-servlets-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [jetty-security-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [jetty-servlet-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]
Caused by: java.sql.SQLSyntaxErrorException: Expression #20 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'cattle.ip_address_1.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
	at org.mariadb.jdbc.internal.util.ExceptionMapper.get(ExceptionMapper.java:125) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.internal.util.ExceptionMapper.throwException(ExceptionMapper.java:69) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.executeQueryEpilog(MariaDbStatement.java:261) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:290) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.MariaDbClientPreparedStatement.execute(MariaDbClientPreparedStatement.java:148) ~[mariadb-java-client-1.3.4.jar:na]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.execute(DelegatingPreparedStatement.java:172) ~[commons-dbcp-1.4.jar:1.4]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.execute(DelegatingPreparedStatement.java:172) ~[commons-dbcp-1.4.jar:1.4]
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:194) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:261) ~[jooq-3.3.0.jar:na]
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:322) ~[jooq-3.3.0.jar:na]
	... 60 common frames omitted
Caused by: org.mariadb.jdbc.internal.util.dao.QueryException: Expression #20 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'cattle.ip_address_1.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:475) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.result(AbstractQueryProtocol.java:396) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:378) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:334) ~[mariadb-java-client-1.3.4.jar:na]
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:280) ~[mariadb-java-client-1.3.4.jar:na]
	... 66 common frames omitted

**************************************
**************************************

Also in the logs are some other errors, not sure if they are related. 

2016-02-16 20:10:37,227 ERROR [4aea62ec-86c3-4167-9c7e-22e4dde24b47:1691] [instance:23] [instance.start->(InstanceStart)->instance.allocate->(InstanceAllocate)] [] [cutorService-24] [c.p.a.e.i.AllocatorEventListenerImpl] No allocator handled [EventVO [id=5f1e8656-833c-4e23-bce2-2be55eb21b52, name=instance.allocate, previousNames=null, replyTo=reply.4973713860490724450, resourceId=23, resourceType=instance, publisher=null, transitioning=null, transitioningMessage=null, transitioningInternalMessage=null, previousIds=null, data={}, time=Tue Feb 16 20:10:37 UTC 2016, listenerKey=null, transitioningProgress=null]] 

2016-02-16 20:10:37,232 ERROR [4aea62ec-86c3-4167-9c7e-22e4dde24b47:1691] [instance:23] [instance.start->(InstanceStart)->instance.allocate] [] [ecutorService-3] [c.p.e.p.i.DefaultProcessInstanceImpl] Unknown exception io.cattle.platform.eventing.exception.EventExecutionException: Scheduling failed: No candidates available
	at io.cattle.platform.eventing.exception.EventExecutionException.fromEvent(EventExecutionException.java:53) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.eventing.impl.AbstractEventService.callSync(AbstractEventService.java:263) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.process.common.handler.EventBasedProcessHandler.handle(EventBasedProcessHandler.java:109) ~[cattle-iaas-logic-common-0.5.0-SNAPSHOT.jar:na]

**Comment by cjellick** [comment]
cross reference: also pointed out in: https://forums.rancher.com/t/rancher-v0-59-0-impossible-to-start-container/1735

**Comment by ibuildthecloud** [comment]
@dbuschman7 Can you tell me what version of MySQL you are running?

**Comment by cjellick** [comment]
We should enforce `sql_mode=only_full_group_by` in the mysql that is built into ranche-server and build-cattle containers so that CI will catch errors like this.

**Comment by cjellick** [comment]
also, we'll fix the query so it doesnt blow up

**Comment by egwada** [comment]
hi !

I confirm what @cjellick reports about **sql_mode=only_full_group_by** ! I change this variable of my MariaDB and it works now !

I check the value for sql_mode

```
MariaDB [(none)]> SELECT @@sql_mode;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
| @@sql_mode                                                                                                                                                      |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,NO_AUTO_VALUE_ON_ZERO,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

So, I set the value for sql_mode like that

```
MariaDB [(none)]> set sql_mode='NO_AUTO_VALUE_ON_ZERO,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
Query OK, 0 rows affected (0.01 sec)

```

I check again and connect to Rancher : my containers can start now.

Thank a lot !

**Comment by dbuschman7** [comment]
@ibuildthecloud Sorry for the delay. I installed the official container, mysql/mysql-server:latest on Feb 6th, 2016. Looks like that would be 5.7.

**Comment by dbuschman7** [comment]
@ibuildthecloud Running the sql_mode command above also get me running and deploying containers. :) Thanks.

**Comment by cjellick** [comment]
Fixed by the above two PRs. I can't think of great way to test this directly. The idea is now that we've turned on the sql mode, cattle drone builds will fail if we write a query that violates it.

**Comment by deniseschannon** [comment]
@dbuschman7 @egwada With v0.63.0 release, could you please test and let us know if it works?

**Comment by deniseschannon** [comment]
@dbuschman7 @egwada If you are still facing issues with this, please re-open. This has been fixed as of v0.63.0.

**Comment by egwada** [comment]
It's ok.We are not facing this problem any more with the V0.63.1 version
